ql/src/java/org/apache/hadoop/hive/ql/parse/DDLSemanticAnalyzer.java

```diff
@@ -2741,7 +2741,7 @@ private void analyzeAlterTableRenamePart(ASTNode ast, String tblName,
     partSpecs.add(oldPartSpec);
     partSpecs.add(newPartSpec);
     addTablePartsOutputs(tab, partSpecs, WriteEntity.WriteType.DDL_EXCLUSIVE);
-    RenamePartitionDesc renamePartitionDesc = new RenamePartitionDesc(tblName, oldPartSpec, newPartSpec);
+    RenamePartitionDesc renamePartitionDesc = new RenamePartitionDesc(tblName, oldPartSpec, newPartSpec, null);
```

Comment:
Sorry, I was wrong about this point earlier. We set the proper current_state_id in ReplicationSpec for each object. Also, we propagate the same ID in the load side as well. So, there isn't any issue.